src/main/java/com/conveyal/datatools/manager/jobs/AutoDeployJob.java

```diff
@@ -0,0 +1,188 @@
+import com.conveyal.datatools.manager.models.OtpServer;
+import com.conveyal.datatools.manager.models.Project;
+import com.conveyal.datatools.manager.persistence.Persistence;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Auto deploy new feed version to OTP server if {@link Project#autoDeploy} is enabled and other conditions are met
+ * (e.g., feed version has no critical errors, active deployment is not in progress, there are no other feed fetches in
+ * progress, etc.). This job must run after {@link ValidateFeedJob} as it has a dependency on the outcome of
+ * {@link FeedVersion#hasCriticalErrors}.
+ */
+public class AutoDeployJob extends MonitorableJob {
+    public static final Logger LOG = LoggerFactory.getLogger(AutoDeployJob.class);
+
+    /**
+     * Projects to be considered for auto deployment.
+     */
+    private List<Project> projects = new ArrayList<>();
+
+    /**
+     * A set of projects which have been locked by a instance of {@link AutoDeployJob} to prevent repeat
+     * deployments.
+     */
+    private static final Set<String> lockedProjects = Collections.synchronizedSet(new HashSet<>());
+
+    /**
+     * Auto deploy specific project.
+     */
+    public AutoDeployJob(Project project, Auth0UserProfile owner) {
+        super(owner, "Auto Deploy Feed", JobType.AUTO_DEPLOY_FEED_VERSION);
+        projects.add(project);
+    }
+
+    /**
+     * Auto deploy all projects.
+     */
+    public AutoDeployJob(Auth0UserProfile owner) {
+        super(owner, "Auto Deploy Feed", JobType.AUTO_DEPLOY_FEED_VERSION);
+        projects = Persistence.projects.getAll();
+    }
+
+    @Override
+    public void jobLogic() {
+        for (Project project : projects) {
+            Deployment deployment = Persistence.deployments.getById(project.pinnedDeploymentId);
+
+            // Define if project, deployment and feed versions are candidates for auto deploy.
+            if (!project.autoDeploy ||
+                project.pinnedDeploymentId == null ||
+                deployment == null ||
+                lockedProjects.contains(project.id) ||
+                project.retrieveProjectFeedSources().isEmpty()) {
+                LOG.info("Project {} skipped for auto deployment as required criteria not met.", project.name);
+                continue;
+            }
+
+            try {
+                LOG.info("Auto deploy task running for project {}", project.name);
+                synchronized (lockedProjects) {
+                    if (!lockedProjects.contains(project.id)) {
+                        lockedProjects.add(project.id);
+                    }
+                }
+
+                if(!canAutoDeployProject(project, deployment, project.retrieveProjectFeedSources())) {
```

Comment:
Sending over all of the feed sources from the project is going to ignore the current deployment configuration's feed sources. Only feed versions within feed sources that have been added to a deployment configuration should be used.

Reply:
I have updated to use the feed versions associated with a deployment i.e. `Deployment.feedVersionIds`.

Reply:
Have addressed in the latest commit.

Reply:
Ended up being addressed in Evans PR. Closing.